### PR TITLE
Fix FindASSIMP.cmake and improve DEAL_II_DISABLE_EXTRA_DIAGNOSTICS

### LINKED ISSUE #5145: Multiple issues with assimp cmake configure
when a user does not specify `ASSIMP_DIR`, but still has its prefix in `CMAKE_PREFIX_PATH`, deal.II is not being configured properly:
```
#        DEAL_II_WITH_ASSIMP set up with external dependencies
#            ASSIMP_INCLUDE_DIRS = /include
#            ASSIMP_USER_INCLUDE_DIRS = /include
#            ASSIMP_LIBRARIES = /Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-9.0.0-apple/assimp-4.0.1-46mnxyqdm5lingdqgg3uk2nmozqtia4z/lib/libassimp.dylib
```

### PR DESCRIPTION
**Pull request by tamiko**
Closes #5145

### CODE REVIEW

File: include/deal.II/base/config.h.in [hunk 1]
```diff
@@ -277,40 +277,42 @@
...
-_Pragma("GCC diagnostic ignored \"-Wpragmas\"")                  \
-_Pragma("GCC diagnostic ignored \"-Wunknown-warning-option\"")   \
-_Pragma("GCC diagnostic ignored \"-Wunknown-warning\"")          \
-_Pragma("GCC diagnostic ignored \"-Wextra\"")                    \
-_Pragma("GCC diagnostic ignored \"-Woverloaded-virtual\"")       \
-_Pragma("GCC diagnostic ignored \"-Wunused-function\"")          \
-_Pragma("GCC diagnostic ignored \"-Wunused-parameter\"")         \
-_Pragma("GCC diagnostic ignored \"-Wunused-variable\"")          \
-_Pragma("GCC diagnostic ignored \"-Wtype-limits\"")              \
+/* keep pragmas with an exclamation mark in order.. */
```
Comment (drwells) [inline include/deal.II/base/config.h.in]: Thank you for adding this not-obvious comment 👍

File: include/deal.II/base/config.h.in [hunk 2]
```diff
@@ -277,40 +277,42 @@
...
+_Pragma("GCC diagnostic ignored \"-Wimplicit-fallthrough\"")            \
+_Pragma("GCC diagnostic ignored \"-Winfinite-recursion\"")              \
+_Pragma("GCC diagnostic ignored \"-Wint-in-bool-context\"")             \
+_Pragma("GCC diagnostic ignored \"-Wmisleading-indentation\"")          \
+_Pragma("GCC diagnostic ignored \"-Wmissing-field-initializers\"")      \
+_Pragma("GCC diagnostic ignored \"-Wnested-anon-types\"")               \
+_Pragma("GCC diagnostic ignored \"-Wnon-virtual-dtor\"")                \
+_Pragma("GCC diagnostic ignored \"-Woverflow\"")                        \
+_Pragma("GCC diagnostic ignored \"-Woverloaded-virtual\"")              \
+_Pragma("GCC diagnostic ignored \"-Wpedantic\"")                        \
```
Comment (drwells) [inline include/deal.II/base/config.h.in]: This is very pedantic but would you mind aligning the `\`s? If you don't I can write a follow-up patch. The wobbly back-slashes bother me more than they should.

**Comment by drwells** [comment]
I would prefer to have a different branch name since this one implies (falsely) that deal.II is not currently great.

File: include/deal.II/base/config.h.in (hunk 2 above)
Comment (bangerth) [reply include/deal.II/base/config.h.in]: We're all type-A personalities, @drwells! :-)

**Pull request merged**